homeassistant/components/backup/manager.py

```diff
@@ -896,7 +852,28 @@ async def _async_create_backup(
                 protected=password is not None,
                 size=size_in_bytes,
             )
-            return (backup, tar_file_path)
+
+            async_add_executor_job = self._hass.async_add_executor_job
+
+            async def send_backup() -> AsyncIterator[bytes]:
+                f = await async_add_executor_job(tar_file_path.open, "rb")
+                try:
+                    while chunk := await async_add_executor_job(f.read, 2**20):
+                        yield chunk
+                finally:
+                    await async_add_executor_job(f.close)
+
+            async def open_backup() -> AsyncIterator[bytes]:
+                return send_backup()
+
+            async def remove_backup() -> None:
```

Comment:
I'm contemplating if we still need the `remove_backup` callback or if we can simply remove the file when the stream ends in the finally block (optionally)?

Reply:
I guess we can't remove it if the stream should be used multiple times.

Reply:
Exactly, we need to handle the case where we stream the backup to multiple agents